src/perform_step/firk_perform_step.jl

```diff
@@ -327,13 +327,13 @@ end
     cache.iter = iter
 
     @. u = uprev + z2
+    step_limiter!(u, f, p, t + dt)
     if adaptive
```

Comment:
```suggestion
    step_limiter!(u, f, p, t + dt)

    if adaptive
```
To keep the separation of the parts as in the original code?